express/code/blocks/banner/banner.js

```diff
@@ -15,6 +15,84 @@ export default async function decorate(block) {
   const isBannerStandoutVariant = block.classList.contains('standout');
   const isBannerCoolVariant = block.classList.contains('cool');
 
+  // Check for background image classes
+  const backgroundImageClasses = [
+    'light-bg',
+    'blue-green-pink-bg',
+    'blue-bg',
+    'blue-pink-orange-bg',
+    'green-blue-red-bg',
+    'blue-purple-gray-bg',
+    'yellow-pink-blue-bg',
+  ];
+
+  let hasBackgroundImage = false;
+  for (const className of backgroundImageClasses) {
+    if (block.classList.contains(className)) {
+      hasBackgroundImage = true;
+      break;
+    }
+  }
+
+  // Load background variants CSS if needed
+  if (hasBackgroundImage) {
+    // Completely disable ALL banner.css files to prevent CSS regression
+    const allBannerCSS = document.querySelectorAll('link[href*="banner.css"]');
+    allBannerCSS.forEach((link) => {
+      link.disabled = true;
+      link.style.display = 'none';
+    });
+
+    // Also disable any inline styles that might interfere
+    const bannerBlocks = document.querySelectorAll('.banner');
+    bannerBlocks.forEach((bannerBlock) => {
+      if (backgroundImageClasses.some((cls) => bannerBlock.classList.contains(cls))) {
+        // Remove any inline styles that might interfere
+        bannerBlock.style.removeProperty('background-color');
+        bannerBlock.style.removeProperty('background');
+        bannerBlock.style.removeProperty('color');
+      }
+    });
+
+    // Load our isolated CSS file
+    if (!document.querySelector('link[href*="banner-bg.css"]')) {
+      const link = document.createElement('link');
+      link.rel = 'stylesheet';
+      link.href = '/express/code/blocks/banner/banner-bg.css';
+      document.head.appendChild(link);
+    }
+  }
+
+  // Create background container for banners with background images
+  if (hasBackgroundImage) {
+    const backgroundContainer = createTag('div', { class: 'background-container' });
+
+    // Move all existing content into the background container
+    while (block.firstChild) {
+      backgroundContainer.appendChild(block.firstChild);
+    }
+
+    block.appendChild(backgroundContainer);
+  }
+
+  // Normalize headings first to ensure proper structure
+  normalizeHeadings(block, ['h2', 'h3', 'h4']);
```

Comment:
Move this back to after the background container creation (line ~111) to match the original execution order. Headings should be normalized after the final DOM structure is established, not before content is moved into background containers.